src/main/java/org/fungover/mmotodo/DatabaseInitializer.java

```diff
@@ -0,0 +1,43 @@
+package org.fungover.mmotodo;
+
+import org.fungover.mmotodo.tag.Tag;
+import org.fungover.mmotodo.tag.TagRepository;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DatabaseInitializer implements ApplicationRunner {
+
+    private final TagRepository tagRepository;
+
+    public DatabaseInitializer(TagRepository tagRepository) {
+        this.tagRepository = tagRepository;
+    }
+
+    @Override
+    public void run(ApplicationArguments args) {
+        if (tagRepository.count() == 0) {
+            Tag tag1 = new Tag();
+            tag1.setName("Design");
+            tag1.setDescription("Paint a pretty picture");
```

Comment:
When creating a new task, as it is right now, if no tagId is provided it will default to tagId = 1. Maybe you could create a "default"/"none" tag here that all tasks defaults to, instead of running the data.sql file? What are ya thoughts?

Reply:
I could, but wouldn't it make more sense to not assign a tagid if there are no tags for the task?

Reply:
So just setting tagId to null? Isn't null values something to avoid when storing things in databases?

Reply:
That is true, but creating a default tag that does nothing doesn't make much sense in my book. I think a better solution would be to create the "link" from the tags to the task instead of the other way around.

Reply:
I agree, that sounds like a nice solution! Then I have nothing more to add, nice job! :)